### PR TITLE
openh264[loongarch]: Add SIMD optimization.

### PR DESCRIPTION
**Pull request by wangluls**

### CODE REVIEW

**Comment by wangluls** [comment]
Hi，
   We've optimized series functions for encoding and decoding. It's effective in performance on loongarch platform. Welcome to review it.

Thanks.

File: codec/encoder/core/loongarch/quant_lsx.c [hunk 1]
```diff
@@ -41,9 +41,52 @@
...
+    vec_pDct20 = __lsx_vxor_v(vec_iSign2, vec_pDct2);
+    vec_pDct20 = __lsx_vsub_w(vec_pDct20, vec_iSign2);
+    vec_pDct21 = __lsx_vadd_w(vec_pFF2, vec_pDct20);
+    vec_pDct21 = __lsx_vmul_w(vec_pDct21, vec_pMF2);
+    vec_pDct21 = __lsx_vsrai_w(vec_pDct21, 16);
+    vec_pDct22 = __lsx_vxor_v(vec_iSign2, vec_pDct21);
+    vec_pDct22 = __lsx_vsub_w(vec_pDct22, vec_iSign2);
+
+    vec_pDct = __lsx_vpickev_h(vec_pDct22, vec_pDct12);
+    __lsx_vst(vec_pDct, pDct, 0);
```
Comment (yinshiyou) [inline codec/encoder/core/loongarch/quant_lsx.c:85]: use __lsx_vstx

File: codec/common/loongarch/intra_pred_com_lsx.c [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  int32_t iStridex15 = (kiStride << 4) - kiStride;
+  int32_t iPredStride = 16;
+  int32_t iPredStridex15 = 240; //(iPredStride<<4)-iPredStride;
+  uint8_t i = 15;
+  __m128i kuiV64_vec;
+
+  do {
+    const uint8_t kuiSrc8 = pRef[iStridex15 - 1];
+    const uint64_t kuiV64 = (uint64_t) (0x0101010101010101ULL * kuiSrc8);
+    kuiV64_vec = __lsx_vreplgr2vr_d(kuiV64);
```
[the remark this answers is not in the document]
Comment (wangluls) [reply codec/common/loongarch/intra_pred_com_lsx.c]: done

File: codec/common/loongarch/deblock_lsx.c [hunk 3]
```diff
@@ -682,3 +683,735 @@ void DeblockLumaEq4V_lsx(uint8_t *pPix, int32_t iStride, int32_t iAlpha,
   __lsx_vstx(q1, pPix, iStride);
   __lsx_vstx(q2, pPix, iStride_x2);
 }
+
+void DeblockLumaEq4H_lsx (uint8_t* pPix, int32_t iStrideY, int32_t iAlpha, int32_t iBeta) {
```
[the remark this answers is not in the document]
Comment (wangluls) [reply codec/common/loongarch/deblock_lsx.c:687]: done

File: codec/common/loongarch/deblock_lsx.c [hunk 4]
```diff
@@ -682,3 +683,735 @@ void DeblockLumaEq4V_lsx(uint8_t *pPix, int32_t iStride, int32_t iAlpha,
...
+  DUP4_ARG2(__lsx_vilvl_b, zero, p0, zero, p1, zero, p2, zero, q0,
+            p0_l, p1_l, p2_l, q0_l);
+  DUP4_ARG2(__lsx_vilvh_b, zero, p0, zero, p1, zero, p2, zero, q0,
+            p0_h, p1_h, p2_h, q0_h);
+  DUP2_ARG2(__lsx_vilvl_b, zero, q1, zero, q2, q1_l, q2_l);
+  DUP2_ARG2(__lsx_vilvh_b, zero, q1, zero, q2, q1_h, q2_h);
+  DUP2_ARG2(__lsx_vilvl_b, zero, p3, zero, q3, p3_l, q3_l);
+  DUP2_ARG2(__lsx_vilvh_b, zero, p3, zero, q3, p3_h, q3_h);
+
+  t0_l = __lsx_vslli_h(p1_l, 1);
```
[the remark this answers is not in the document]
Comment (wangluls) [reply codec/common/loongarch/deblock_lsx.c:742]: done

**Comment by wangluls** [comment]
There are conflicts between PR 3492 and 3493, so close 3492 and append it to 3493. Meson build support for loongarch is also appended . Thanks.

**Comment by yinshiyou** [comment]
looks good to me.

**Comment by GuangweiWang** [comment]
Looks good to me. 
just have a suggestion, It is better to add the corresponding UT for all functions. :)

**Pull request merged**